client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -39,14 +44,44 @@ export class SimpleSiteRenameForm extends Component {
 		this.props.requestSiteRename( selectedSiteId, this.state.domainFieldValue, discard );
 	};
 
+	getDomainValidationMessage( domain ) {
+		const { translate } = this.props;
+
+		if ( isEmpty( domain ) ) {
+			return '';
+		}
+
+		if ( domain.match( /[^a-z0-9]/i ) ) {
```

Comment:
I wonder if it would just be annoying to reject any capitalisation in the field though. I may want to enter `SpencerTaylor` but would see errors. If we do go this route we'd need to be sure to add another validation message to specifically inform the user as they type that capitals are invalid.

Maybe a good balance would be to make sure that we show the new domain in lowercase, as right now we still communicate to the customer that their domain name will be capitalised:

<img width="300" alt="screen shot 2018-02-19 at 11 12 59" src="https://user-images.githubusercontent.com/4335450/36361354-3b222554-1566-11e8-9d34-9c64adba8621.png">

Would become:

<img width="306" alt="screen shot 2018-02-19 at 11 16 01" src="https://user-images.githubusercontent.com/4335450/36361369-4bc60cd6-1566-11e8-9676-b467e4c4d746.png">

Regardless of the input capitalisation.

Reply:
Sorry, I was too vague - yes, that's what I thought too, we should lowercase the name for the user, maybe even on the input stage.

Reply:
This is handled indirectly by forcing the input value to lowercase in https://github.com/Automattic/wp-calypso/pull/22204/commits/6f1ddb468672649bb7d34005a75119b4b68198aa :)

Reply:
I took the `lowercase` out of the error message. It's implied by the range in parentheses & we're auto-translating from upper to lower in the client.